src/file.cpp

```diff
@@ -1111,6 +1149,30 @@ namespace libtorrent
 		remove(f, ec);
 	}
 
+	void remove_empty(std::string const& f, error_code& ec)
+	{
+		ec.clear();
+
+		file_status s;
+		stat_file(f, &s, ec);
+		if (ec) return;
+
+		if (s.mode & file_status::directory)
+		{
+			for (directory i(f, ec); !i.done(); i.next(ec))
+			{
+				if (ec) return;
+
+				std::string p = i.file();
+				if (p == "." || p == "..") continue;
+				remove_empty(combine_path(f, p), ec);
+				ec.clear();
+			}
+
+			remove(f, ec);
```

Comment:
Honestly, I could not think that there may be empty directories, not belonging to the torrent. But we can't say that they can not be. So I'll fix it. The only assumption that I have to do, if the directory belongs to torrent and it does not contain any more "foreign" files, we will delete it (we can't know the backstory).
And now I no longer need `remove_empty()` function, so I'll get rid of it.